### ISSUE TITLE
Keras

### ISSUE THREAD

**Issue by benam2** [akashgit/Neural-Variational-Inference-for-Topic-Models#17]
Hi,

Again thanks for sharing your code. You linked to the keras version of this implementation. Are you sure that version is valid?

I went through it and I dont know why the way he/she calculated `mean` and `sigma` is the same. Could you please comment on this?

The Keras repository is read only thats why I could not open an issue there. I just want to know if you think the same as me that the implementation is not correct or am I missing somethin.

`x = Input(batch_shape=(batch_size, V))
h = Dense(num_hidden, activation='softplus')(x)
h = Dense(num_hidden, activation='softplus')(h)
z_mean = BatchNormalization()(Dense(num_topic)(h))
z_log_var = BatchNormalization()(Dense(num_topic)(h))`

Thanks for taking the time.

**Comment by nzw0301** [comment]
Your mentioned part in my old keras code is almost the same as the original implementation that is defined in [these lines](https://github.com/akashgit/autoencoding_vi_for_topic_models/blob/master/models/prodlda.py#L89-L101). I believe that the difference is only the existence of dropout, which I have missed.

If you have a question for my repo, please send e-mail to me.

**Comment by benam2** [comment]
Thanks a lot @nzw0301 for the quick reply. (I want to use the Keras code that's why it's important to me to understand it well).

I will review it one more time to make sure I am not missing anything and then will email you.

Thanks again ~
